### ISSUE TITLE
Improve scoping of fields for examples

### ISSUE THREAD

**Issue by nigelcharman** [concordion/concordion#142]
As per the [discussion on concordion-dev](https://groups.google.com/forum/#!topic/concordion-dev/jRR0tUxZ96U), some work needs to be done to align Concordion with JUnit, and create a new instance of the Java class per example.

This will cause some people issues if they need shared scope per specification that is not static - eg. expensive shared state.

Two changes are needed:

1. Create a new Java instance for each example
2. Find a way to share specification state across examples

**Comment by nigelcharman** [comment]
Some initial discussion with Tim and Andrew:

One potential implementation:

    @ConcordionScope(Specification)

possibly initialised using methods
    @BeforeSpecification
    @AfterSpecifiication

**Comment by nigelcharman** [comment]
Or as a JUnit rule:

```
@Rule public ConcordionSharedObject<Browser.class> cso = new ...

cso.getObj();
```

**Comment by nigelcharman** [comment]
May also need a BeforeSpec example scope in the spec

**Comment by drtimwright** [comment]
I've created a pull request for this. https://github.com/concordion/concordion/pull/145

**Comment by nigelcharman** [comment]
Implemented in Concordion [2.0.0-SNAPSHOT](https://github.com/concordion/concordion/releases/tag/2.0.0-SNAPSHOT-20160125).
